cms/taxonomy/models.py

```diff
@@ -0,0 +1,114 @@
+from wagtail.admin.panels import FieldPanel
+from wagtail.search import index
+
+BASE_TOPIC_DEPTH = 2
+
+
+class TopicManager(models.Manager):
+    def get_queryset(self) -> QuerySet:
+        """Filter out the root topic from all querysets."""
+        return super().get_queryset().filter(depth__gt=1)
+
+    def root_topic(self) -> "Topic":
+        return typing.cast(Topic, super().get_queryset().filter(depth=1).first())
+
+
+# This is the main 'node' model, it inherits mp_node
+# mp_node is short for materialized path, it means the tree has a clear path
+class Topic(index.Indexed, MP_Node):
+    """A topic model, representing taxonomic topics.
+    We use tree nodes to represent the topic/subtopic parent/child relationships.
+
+    Note:
+    We must be able to cope with topics potentially moving to and from root level. However, Nodes cannot be moved from
+    root level in treebeard. To cope with this, we put all topics underneath a dummy root level node. To hide this
+    dummy node, we override the default object manager with one which only returns non-root level, actual topic nodes.
+    """
+
+    class Meta:
+        ordering = ("path", "title")
+
+    objects: TopicManager = TopicManager()  # Override the default manager
+
+    id: ClassVar = models.CharField(max_length=100, primary_key=True)
+    title: ClassVar = models.CharField(max_length=100)
+    description: ClassVar = models.TextField(blank=True, null=True)
+    removed: ClassVar = models.BooleanField(default=False)
+
+    node_order_by: ClassVar[list[str]] = ["title"]
+
+    search_fields: ClassVar[list[index.SearchField | index.AutocompleteField]] = [
+        index.FilterField("title"),
+        index.FilterField("depth"),
+        index.SearchField("title"),
+        index.AutocompleteField("title"),
+    ]
+
+    def save_topic(self, parent_topic: Optional["Topic"] = None) -> None:
+        """Save a topic either underneath the specific parent if passed, otherwise underneath our default root level
+        dummy topic.
+        """
+        if not parent_topic:
+            parent_topic = Topic.objects.root_topic()
+        parent_topic.add_child(instance=self)
+        super().save()
+        parent_topic.save()
+
+    def get_parent(self, *args: Any, **kwargs: Any) -> Optional["Topic"]:
+        """Return the parent topic if one exists, or None otherwise.
+        Return none if at or below our base topic depth to avoid returning a cached root topic.
+        """
+        if self.depth <= BASE_TOPIC_DEPTH:
+            return None
+        return super().get_parent(*args, **kwargs)
+
+    def move(self, target: Optional["Topic"] = None, **kwargs: Any) -> None:  # pylint: disable=arguments-differ
+        """Move the topic to underneath the target parent. If no target is passed, move it underneath our root."""
+        if not target:
+            return super().move(Topic.objects.root_topic(), **kwargs)
+        return super().move(target, **kwargs)
```

Comment:
Minor: If `move` doesn't return anything you don't need `return` statement. Also, it could remove the need for the two calls.
```py
super().move(target if target else Topic.objects.root_topic(), **kwargs)
```

or if clearer:
```py
parent = target or Topic.objects.root_topic()
super().move(parent, **kwargs)
```

Reply:
Addressed in https://github.com/ONSdigital/dis-wagtail/pull/104/commits/89fb9c9c8c63153bba489e9abedc3084ec285310